addons/mail/controllers/gif.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route, Controller
+import requests
+import json
+
+
+class MailGifController(Controller):
+
+    ckey = "my_test_app"
+    limit = 8
+    # This is the equivalent of PG-13
+    # see https://developers.google.com/tenor/guides/content-filtering
+    contentFilter = "low"
+    tenor_url = "https://tenor.googleapis.com"
+
+    @route("/mail/gif/search", type="json", auth="user", cors="*")
+    def mail_gif_search(self, search_term, locale="en", country="US", pos=False):
+        api_key = request.env["ir.config_parameter"].sudo().get_param("mail.tenor_api_key")
+        url = f"{self.tenor_url}/v2/search?q={search_term}&key={api_key}&client_key={self.ckey}&limit={self.limit}&contentfilter={self.contentFilter}&locale={locale}&country={country}&media_filter=tinygif"
+        if pos:
+            url = f"{url}&pos={pos}"
+        r = requests.get(url)
+        return json.loads(r.content)
+
+    @route("/mail/gif/categories", type="json", auth="user", cors="*")
+    def mail_gif_categorie(self, locale="en", country="US"):
+        api_key = request.env["ir.config_parameter"].sudo().get_param("mail.tenor_api_key")
+        url = f"{self.tenor_url}/v2/categories?key={api_key}&client_key={self.ckey}&contentfilter={self.contentFilter}&locale={locale}&country={country}"
+        r = requests.get(url)
+        return json.loads(r.content)
+
+
+    @route("/mail/gif/set_favorite", type="json", auth="user")
+    def set_gif_favorite(self, gif_id):
+        request.env["mail.gif_favorite"].create({"gif_id": gif_id})
+
+    def mail_gif_posts(self, ids):
+        api_key = request.env["ir.config_parameter"].sudo().get_param("mail.tenor_api_key")
+        url = f"{self.tenor_url}/v2/posts?key={api_key}&client_key={self.ckey}&ids={','.join(ids)}&media_filter=tinygif"
+        r = requests.get(url)
+        return json.loads(r.content)["results"]
+
+    @route("/mail/gif/favorites", type="json", auth="user")
+    def get_gif_favorites(self, offset=0):
+        gif_ids = request.env["mail.gif_favorite"].search([("create_uid", "=", request.env.user.id)], limit=20, offset=offset)
+        return {
+            "results": self.mail_gif_posts(gif_ids.mapped("gif_id")) or [],
+            "offset": offset + 20
+        }
+
+    @route("/mail/gif/remove_favorite", type="json", auth="user")
+    def remove_gif_favorite(self, gif_id):
+        return request.env["mail.gif_favorite"].search([("gif_id", "=", gif_id)]).unlink()
```

Comment:
You are removing it for all users, which seem counter intuitive as the search is user-based.